packages/client/src/pages/Title.re

```diff
@@ -0,0 +1,42 @@
+[@react.component]
+let make = (~route: Routing.Route.t) => {
+  switch (route) {
+  | Channel({channelId, subPage}) =>
+    ChannelGet.component2(~id=channelId, result =>
+      switch (result) {
+      | Success(channel) =>
+        let title = [|channel.name, Lang.Title.main|] |> E.Title.toString;
+        <ForetoldComponents.Title title />;
+      | _ => <Null />
+      }
+    )
+  | ChannelIndex =>
+    <ForetoldComponents.Title
```

Comment:
I think some more dryness / functional additions would be nice.

Maybe one function that returns an Option(string) with the title, and another that wraps that with <ForetoldComponents.Title/>

Reply:
No, we should use Factories this way. Only components.

Reply:
I have done because of loading channel data with components. I do not want to divide "switch" construction on two and do it in a complicated way. It is not readable.

Reply:
Although it should be cleaned. Will think about it later.

Reply:
If nothing is quickly obvious, don't worry about it.

Reply:
Yes, please let us leave this for now later I will fix this when I find some useful pattern.